### DISCUSSION TITLE
Couldn't understand the need of smart contracts.

### DISCUSSION THREAD

**Discussion by naveenburathi** [smartcontractkit/full-blockchain-solidity-course-py#593] [Q&A]
Hlo everyone,
I have reached to 6th lesson of the course. 
So I was thinking that a project like Lottery can be made by normal web development frameworks. So what is the need of the smart contracts since it can be made with normal backend. What are the lackings in normal web development frameworks and database?
I know this is a silly query and there would be a strong reason behind this. But I couldn't answer this query for myself.    
Looking for resolving this silly doubt.

**Comment by cromewar** [comment]
Hello @naveenburathi  well this is actually a hard question, because you are right, this already could be done in the normal web2 way. So the point of doing this as we do on the course is more for a conceptual/philosophical reason, and there is some points to get in consideration:

1. Decentralization: we use an admin on the course just for simplicity purposes, but a  this kind of lotteries are usually governed by a decentralized oracle.
2. anonymity: User wont have to use their real names, or their debit cards with their info. The lottery is executed on the blockchain so the users are pseudo anonymous.
3. Also there are the advantages of the blockchain itself, immutability ,security and more....
4. If you refer to the first section of the course, Patrick goes in deep why using smart contracts is better in general for any agreement.

I hope this might help.
